client/my-sites/stats/promo-cards/index.jsx

```diff
@@ -2,49 +2,70 @@ import { recordTracksEvent } from '@automattic/calypso-analytics';
 import { MobilePromoCard } from '@automattic/components';
 import { translate } from 'i18n-calypso';
 import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import blazeDropDownIllustration from 'calypso/assets/images/illustrations/blaze-drop-down.svg';
 import wordpressSeoIllustration from 'calypso/assets/images/illustrations/wordpress-seo-premium.svg';
 import PromoCardBlock from 'calypso/blocks/promo-card-block';
 import DotPager from 'calypso/components/dot-pager';
+import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 
 import './style.scss';
 
 const EVENT_YOAST_PROMO_VIEW = 'calypso_stats_wordpress_seo_premium_banner_view';
+const EVENT_BLAZE_PROMO_VIEW = 'calypso_stats_blaze_banner_view';
 const EVENT_MOBILE_PROMO_VIEW = 'calypso_stats_traffic_mobile_cta_jetpack_view';
 
-export default function PromoCards( { isJetpack, isOdysseyStats, slug } ) {
+export default function PromoCards( { isOdysseyStats, slug } ) {
+	const selectedSiteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+	const jetpackNonAtomic = useSelector(
+		( state ) => isJetpackSite( state, selectedSiteId ) && ! isAtomicSite( state, selectedSiteId )
+	);
+
 	// Yoast promo is disabled for Odyssey & self-hosted.
 	// Mobile apps promos are always shown.
-	const showYoastPromo = ! isOdysseyStats && ! isJetpack;
-	// Handle initial view event if not using the DotPager UI.
-	// We don't worry about the Yoast card as it sends on mount.
-	useEffect( () => {
-		if ( ! showYoastPromo ) {
-			recordTracksEvent( EVENT_MOBILE_PROMO_VIEW );
-		}
-	}, [ showYoastPromo ] );
+	const showBlazePromo = ! isOdysseyStats;
+	const showYoastPromo = ! isOdysseyStats && ! jetpackNonAtomic;
+
 	// Handle click events from promo card.
 	const promoCardDidReceiveClick = ( event ) => {
 		// Events need to incorporate the page and the click type.
-		// This will allow us to account for integraton across different pages.
+		// This will allow us to account for integration across different pages.
 		// FORMAT: calypso_stats_PAGE_mobile_cta_EVENT
 		const tracksEventName = `calypso_stats_traffic_mobile_cta_${ event.replaceAll( '-', '_' ) }`;
 		recordTracksEvent( tracksEventName );
 	};
 	// Handle view events from DotPager UI.
 	const pagerDidSelectPage = ( index ) => {
-		const evenLookup = [ EVENT_YOAST_PROMO_VIEW, EVENT_MOBILE_PROMO_VIEW ];
-		const eventName = evenLookup[ index ];
+		const eventLookup = [ EVENT_BLAZE_PROMO_VIEW, EVENT_YOAST_PROMO_VIEW, EVENT_MOBILE_PROMO_VIEW ];
+		const eventName = eventLookup[ index ];
 		recordTracksEvent( eventName );
 	};
-	// Render one or both promo cards.
+
+	// Trigger initial view event as DotPager don't call it.
+	useEffect( () => pagerDidSelectPage( 0 ) );
```

Comment:
This assumes the DotPager resets to page zero when changing sites which isn't the case.

1. Select a site and view the 2nd page (Yoast).
2. Changing sites will trigger a new Blaze view event despite the card still showing Yoast.

The old code handled this by not sending new impression events when the site changed which seems reasonable. This new behaviour will send the wrong impression events though.

It's a limitation of the DotPager that it doesn't allow us to reset the page so the correct fix might be to track the last selected page as local state and use that, in combo with the `slug`, as dependancies for firing this.

Reply:
Thanks for all the tests and this fix suggestion @a8ck3n!

I tried to fix storing the last selected page as local state combining with slug change but didn't find a good fix in my timebox.

It seems the best way to track this would be to fix DotPager's limitation by resetting the view, so I brought back the old logic of not sending new impressions events when the site changed, as we already sent them on the component mount.